harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,596 @@
+            ),
+        )
+
+
+def validate_accelerator_zone(args: argparse.Namespace, zone: str) -> None:
+    try:
+        subprocess.check_call(
+            [
+                "gcloud",
+                "compute",
+                "accelerator-types",
+                "describe",
+                args.gpu_type,
+                "--zone",
+                zone,
+            ],
+            stdout=subprocess.DEVNULL,
+        )
+    except subprocess.CalledProcessError:
+        raise ValueError(
+            "The specified accelerator type {} was not found in the specified zone {}.".format(
+                args.gpu_type, zone
+            ),
+        )
+
+
+def validate_args(args: argparse.Namespace) -> None:
+    validate_location(args.zone, isZone=True)
+    validate_accelerator_type(args.gpu_type)
+
+    if args.master_machine_type != "n1-standard-16":
+        validate_machine_type(args.master_machine_type, args.zone)
+
+    if args.agent_machine_type != "n1-standard-32":
+        validate_machine_type(args.agent_machine_type, args.zone)
+
+    validate_accelerator_zone(args, args.zone)
+
+    if args.gpu_coscheduler and args.preemption:
+        raise ValueError(
+            "--gpu-coscheduler and --preemptive-scheduler are mutually exclusive and cannot both be"
+            " specified",
+        )
+
+    if args.multi_node_pool and args.cpu_coscheduler and not args.gpu_coscheduler:
+        raise ValueError(
+            "To enable coscheduling on the CPU Node Pool, coscheduling must be enabled on the GPU"
+            " Node Pool. --cpu-coscheduler cannot be specified if --gpu-coscheduler isn't as well."
+        )
+    elif args.cpu_coscheduler and not args.cpu_only:
+        raise ValueError(
+            "CPU co-scheduling can not be enabled on a GPU Node Pool only instance."
+            + " Please specify one of --cpu-only or --multi-node-pool to create a CPU Node Pool."
+        )
+
+    if args.gcs_bucket_name is None:
+        args.gcs_bucket_name = args.cluster_id + "-checkpoints"
+    if args.agent_node_pool_name is None:
+        args.agent_node_pool_name = args.cluster_id + "-gpu-pool"
+    if args.cpu_node_pool_name is None:
+        args.cpu_node_pool_name = args.cluster_id + "-cpu-pool"
+    if not Path(args.helm_dir).exists():
+        raise ValueError("Please specify valid --helm_dir")
+
+
+def create_cluster(args: argparse.Namespace) -> None:
+    region = args.zone.rsplit("-", 1)[0]
+    cmd = [
+        "gcloud",
+        "container",
+        "clusters",
+        "create",
+        args.cluster_id,
+        "--region",
+        region,
+        "--node-locations",
+        args.zone,
+        "--num-nodes=1",
+        "--machine-type={}".format(args.master_machine_type),
+    ]
+    subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+
+    create_nodepools(region, args)
+
+    if not args.no_manage_bucket:
+        cmd = ["gsutil", "mb", "gs://{}".format(args.gcs_bucket_name)]
```

Comment:
Would be nice to see if gsutil, gcloud, etc. is on the PATH and fail with a helpful message if it's missing.

Reply:
I think it'd already give a somewhat understandable message: `FileNotFoundError: [Errno 2] No such file or directory: 'gsutil'`
Or do you want something better?

Reply:
That works. Failing early before we've attempted ANY steps (and perhaps created a partial deployment) is a nice-to-have. It doesn't need to block the PR though.